keras_cv/models/backbones/densenet/densenet_backbone.py

```diff
@@ -0,0 +1,367 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DenseNet backbone model.
+
+Reference:
+  - [Densely Connected Convolutional Networks](https://arxiv.org/abs/1608.06993)
+  - [Based on the Original keras.applications DenseNet](https://github.com/keras-team/keras/blob/master/keras/applications/densenet.py)
+"""  # noqa: E501
+
+import copy
+
+from tensorflow import keras
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.densenet.densenet_backbone_presets import (
+    backbone_presets,
+)
+from keras_cv.models.backbones.densenet.densenet_backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.utils.python_utils import classproperty
+
+BN_AXIS = 3
+BN_EPSILON = 1.001e-5
+
+
+def apply_dense_block(x, blocks, name=None):
+    """A dense block.
+
+    Args:
+      x: input tensor.
+      blocks: int, number of building blocks.
+      name: string, block label.
+
+    Returns:
+      a function that takes an input Tensor representing an apply_dense_block.
+    """
+    if name is None:
+        name = f"dense_block_{backend.get_uid('dense_block')}"
+
+    for i in range(blocks):
+        x = apply_conv_block(x, 32, name=f"{name}_block_{i}")
```

Comment:
Let's expose `growth_rate` as a top level param. Having this const hanging out here is weird